packages/common/src/codegen/render-solidity/common.ts

```diff
@@ -180,6 +180,18 @@ export function renderValueTypeToBytes32(name: string, { typeUnwrap, internalTyp
   }
 }
 
+export function isLeftAligned(field: Pick<RenderType, "internalTypeId">): boolean {
+  return field.internalTypeId.match(/^bytes\d{1,2}$/) !== null;
+}
+
+export function shiftLeftBits(field: Pick<RenderType, "internalTypeId" | "staticByteLength">): number {
```

Comment:
nit, feel like this would be a bit more descriptive
```suggestion
export function getLeftPaddingBits(field: Pick<RenderType, "internalTypeId" | "staticByteLength">): number {
```